CONTRIBUTING.md

```diff
@@ -148,160 +132,166 @@ When updating this, also update:
-options #![cfg] declaration defines for that file.
-
-For instance, to run a test marked with the 'tokio_unstable' cfg option,
-you must pass this flag to the compiler when running the test.
-```
-$ RUSTFLAGS="--cfg tokio_unstable" cargo test -p tokio --all-features --test rt_metrics
+If the change being proposed alters code (as opposed to only documentation for example), it is
+either adding new functionality to Tokio or it is fixing existing, broken functionality. In both of
+these cases, the pull request should include one or more tests to ensure that Tokio does not regress
+in the future. There are two ways to write tests: [integration tests][integration-tests] and
+[documentation tests][documentation-tests]. (Tokio avoids [unit tests][unit-tests] as much as
+possible).
+
+Tokio uses [conditional compilation attributes][conditional-compilation] throughout the codebase, to
+modify rustc's behavior. Code marked with such attributes can be enabled using RUSTFLAGS and
+RUSTDOCFLAGS environment variables. One of the most prevalent flags passed in these variables is the
+`--cfg` option. To run tests in a particular file, check first what options #![cfg] declaration
+defines for that file.
+
+For instance, to run a test marked with the 'tokio_unstable' cfg option, you must pass this flag to
+the compiler when running the test.
+
+```shell
+RUSTFLAGS="--cfg tokio_unstable" cargo test -p tokio --all-features --test rt_metrics
 ```
 
 #### Integration tests
 
-Integration tests go in the same crate as the code they are testing. Each sub
-crate should have a `dev-dependency` on `tokio` itself. This makes all Tokio
-utilities available to use in tests, no matter the crate being tested.
+Integration tests go in the same crate as the code they are testing. Each sub crate should have a
+`dev-dependency` on `tokio` itself. This makes all Tokio utilities available to use in tests, no
+matter the crate being tested.
 
-The best strategy for writing a new integration test is to look at existing
-integration tests in the crate and follow the style.
+The best strategy for writing a new integration test is to look at existing integration tests in the
+crate and follow the style.
 
 #### Fuzz tests
 
-Some of our crates include a set of fuzz tests, this will be marked by a
-directory `fuzz`. It is a good idea to run fuzz tests after each change.
-To get started with fuzz testing you'll need to install
-[cargo-fuzz](https://github.com/rust-fuzz/cargo-fuzz).
+Some of our crates include a set of fuzz tests, this will be marked by a directory `fuzz`. It is a
+good idea to run fuzz tests after each change. To get started with fuzz testing you'll need to
+install [cargo-fuzz](https://github.com/rust-fuzz/cargo-fuzz).
 
-`cargo install cargo-fuzz`
+```shell
+cargo install --locked cargo-fuzz
+```
 
 To list the available fuzzing harnesses you can run;
 
-```bash
-$ cd tokio
-$ cargo fuzz list
+```shell
+cd tokio
+cargo fuzz list
+```
+
+```text
 fuzz_linked_list
 ````
```

Comment:
It's a good idea to generally split commands and the output into separate blocks instead of using shell prefixes, to make it clearer that the user shouldn't be pasting the output of the command.